primitives/src/lib.rs

```diff
@@ -60,6 +60,7 @@ pub mod icicle_deps {
     // TODO: remove this after `warmup()` is added upstream
     // https://github.com/ingonyama-zk/icicle/pull/422#issuecomment-1980881638
     /// Create a new stream and warmup
+    #[allow(clippy::result_unit_err)]
```

Comment:
Can we return an anyhow instead (or PrimitiveError)? We are using that in VId anyway.

Reply:
Done in [e5312bb](https://github.com/EspressoSystems/jellyfish/pull/523/commits/e5312bb28d8e116b74031e3845ca3936704ae046)